### ISSUE TITLE
Does ghci mode recognize cabal sandboxes?

### ISSUE THREAD

**Issue by fosskers** [haskell/haskell-mode#398]
Forgive me if this has been asked before, or if the solution is trivial. I'd like to be able to install libraries in a sandbox, then play with them in an emacs session without making a full cabal project (similar to [this post](http://stackoverflow.com/questions/26756056/ghci-cabal-sandbox-flags-for-emacs-haskell-inferior-mode) on stackoverflow).

Is this possible?

**Comment by fosskers** [comment]
I must be messing up the order somehow. Otherwise I've followed what you did exactly.

`emacs -Q test.emacs.el &` yields `Job 3, “emacs -Q test.emacs.el &” has stopped`

Then, either by just doing `emacs -Q` then visiting the Haskell file, or by `emacs -Q ShellyTest.hs` none of the config (defined in `test.emacs.el` or otherwise) loads. I'm left in Fundamental mode with `M-x eval-buffer` doing nothing.

Any insight?

**Comment by fosskers** [comment]
I've managed to succeed in following your instructions, in that I can open my test file with a minimal config (I can tell because a lot of my customisations are missing) but unfortunately `C-c C-l` still dies with the same error.

**Comment by fosskers** [comment]
Interesting, when I run the program `cabal repl` in a directory without a cabal file I get exactly the text we've been seeing:

```
Package has never been configured. Configuring with default flags. If this
fails, please run configure manually.
cabal: No cabal file found.
Please create a package description file <pkgname>.cabal
```

What do you see?

**Comment by ivan-m** [comment]
Just to double-check: what does `cabal -V` say?  I have `1.20.0.4`.

```
$cabal repl
GHCi, version 7.8.3: http://www.haskell.org/ghc/  :? for help
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Prelude> 
```

**Comment by fosskers** [comment]
The Arch `cabal-install` was at `1.18.x` but I made a sandbox, updated cabal in there (it pulls down other libs to compile, which I didn't want clobbering my global haskell environment), then added it to my path ahead of `/usr/bin`. I'm now on `1.20.0.6`, and everything works. Calling `cabal repl` in the shell opens the repl as it should, and I can load Shelly as desired in emacs.

Note: I had to restart my emacs daemon once to (I assume) get it to "see" the newer cabal binary.
